ncm-dpmlfc/src/main/perl/DPMLFC/RuleBasedEditor.pm

```diff
@@ -0,0 +1,778 @@
+    return undef;
+  }
+  my $line_fmt = shift;
+  unless ( defined($line_fmt) ) {
+    $self->error("$function_name: 'line_fmt' argument missing");
+    return undef;
+  }
+  my $config_value = shift;
+  if ( $config_value ) {
+    $self->debug(2,"$function_name: configuration value '$config_value' will be added to the pattern");
+    $config_value =~ s/\s+/\\s+/g;
+    $config_value =~ s/\-/\\-/g;
+    $config_value =~ s/\./\\./g;
+    $config_value =~ s/\*/\\*/g;
+    $config_value =~ s/\[/\\[/g;
+    $config_value =~ s/\]/\\]/g;
+    $config_value =~ s/\(/[(]/g;
+    $config_value =~ s/\)/\\)/g;
+    $config_value =~ s/\$/\\\$/g;
+    $config_value =~ s/\^/\\^/g;
+  } else {
+    $config_value = "";
+  }
+
+  $config_param =~ s/\s+/\\s+/g;
+  my $config_param_pattern;
+  if ( $line_fmt == LINE_FORMAT_PARAM ) {
+    $config_param_pattern = "#?\\s*$config_param=".$config_value;
+  } elsif ( $line_fmt == LINE_FORMAT_ENVVAR ) {
+    $config_param_pattern = "#?\\s*export $config_param=".$config_value;
+  } elsif ( $line_fmt == LINE_FORMAT_XRDCFG_SETENV ) {
+    $config_param_pattern = "#?\\s*setenv\\s+$config_param\\s*=\\s*".$config_value;
+  } elsif ( $line_fmt == LINE_FORMAT_XRDCFG_SET ) {
+    $config_param_pattern = "#?\\s*set\\s+$config_param\\s*=\\s*".$config_value;
+  } elsif ( $line_fmt == LINE_FORMAT_XRDCFG ) {
+    $config_param_pattern = "#?\\s*$config_param";
+    # Avoid adding a withespace requirement if there is no config_value
+    if ( $config_value ne "" ) {
+      $config_param_pattern .= "\\s+" . $config_value;
+    }
+  } else {
+    $self->error("$function_name: invalid line format ($line_fmt). Internal inconsistency.");
+    return undef;
+  }
+
+  return $config_param_pattern
+}
+
+
+# This function comments out a configuration line matching the configuration parameter.
+# Match operation takes into account the line format.
+#
+# Arguments :
+#        fh : a FileEditor object
+#        config_param: parameter to update
+#        line_fmt : line format (see LINE_FORMAT_xxx constants)
+sub removeConfigLine () {
+  my $function_name = "removeConfigLine";
+  my $self = shift;
+
+  my $fh = shift;
+  unless ( $fh ) {
+    $self->error("$function_name: 'fh' argument missing");
+    return 1;
+  }
+  my $config_param = shift;
+  unless ( $config_param ) {
+    $self->error("$function_name: 'config_param' argument missing");
+    return 1;
+  }
+  my $line_fmt = shift;
+  unless ( defined($line_fmt) ) {
+    $self->error("$function_name: 'line_fmt' argument missing");
+    return 1;
+  }
+
+  # Build a pattern to look for.
+  my $config_param_pattern = $self->buildLinePattern($config_param,$line_fmt);
+
+  $self->debug(1,"$function_name: commenting out lines matching pattern >>>".$config_param_pattern."<<<");
+  # All matching lines must be commented out, except if they are already commented out.
+  # The code used is a customized version of FileEditor::replace() that lacks backreferences.
```

Comment:
Comment clarified. The problem is that here we want to rewrite the same line commented out but we don't know the current line contents, only a regexp matching it. We would need to express the `newvalue` argument in `replace()` whith some kind of backreference to what was matched by the pattern. This would required the ability to retrieve the group values as we can do in Python with `re.group`. I don't know if this is doable in Perl... if it was, could be a useful extension to `replace()`